[CIR] Add integer result type for cir.global_view

Lancern opened the pull request:
This PR updates the `#cir.global_view` attribute and make it accept integer types as its result type.

bcardosolopes commented on clang/test/CIR/Lowering/globals.cir:
```diff
@@ -198,4 +205,17 @@ module {
   // MLIR:  %0 = llvm.mlir.addressof @zero_array
 
+  cir.func @const_global_addr() -> !u64i {
+    %0 = cir.const #cir.global_addr<@".str"> : !u64i
```
Sounds like this is equivalent to a global view with zero index? Why can't this be modeled ash `cir.const #cir.global_view<@".str"> `?

Lancern replied:
cir.global_addr is of integer types, and I'm not sure if it's appropriate to make cir.global_view an integer type. If it's OK I could just update cir.global_view instead of introducing a new one.

bcardosolopes replied:
My take is that we want here is a constant pointer coming out of `cir.const`, followed up by a cast to integer?

Lancern replied:
> My take is that we want here is a constant pointer coming out of cir.const, followed up by a cast to integer?

In this case, yes. But I'm actually proposing this attribute for the following scenario where you don't have space for additional operations:

```mlir
%0 = cir.const #cir.const_struct {
  #cir.global_addr @aaa,
  #cir.global_addr @bbb
}
```

And I'm looking at this scenario because recently I'm working on the LLVM lowering of member function pointers. Upon ABI lowering, a member function pointer is lowered to a struct with two fields of type `ptrdiff_t`. When the member function pointer represents a non-virtual member function, the first field stores the address of the target function _as an integer_. Thus to represent constant member function pointers I need an attribute that works like `#cir.global_addr`.

Lancern commented:
I have updated this PR and removed the `#cir.global_addr` attribute. Instead, I just updated the existing `#cir.global_view` attribute and make it accept integer types as its result type.

Lancern commented on clang/test/CIR/Lowering/globals.cir:
```diff
@@ -27,15 +27,22 @@ module {
   cir.global external @alpha = #cir.const_array<[#cir.int<97> : !s8i, #cir.int<98> : !s8i, #cir.int<99> : !s8i, #cir.int<0> : !s8i]> : !cir.array<!s8i x 4>
   cir.global "private" constant internal @".str" = #cir.const_array<"example\00" : !cir.array<!s8i x 8>> : !cir.array<!s8i x 8> {alignment = 1 : i64}
   cir.global external @s = #cir.global_view<@".str"> : !cir.ptr<!s8i>
+  cir.global external @s_addr = #cir.global_view<@".str"> : !u64i
```
Updated.

Lancern commented:
Rebased onto the latest `main`.

Lancern commented:
I realized that I have the permissions to merge PRs in this repo. Could I merge this PR now as it left no further unresolved reviews?

bcardosolopes commented:
If they are approved and you addressed all nits please go ahead, thanks for asking. Merging it now cause I'm already at it, but feel free to do it next time around
